### ISSUE TITLE
There are Fake FTDI chips around

### ISSUE THREAD

**Issue by mterwoord** [Jark/FTDISample#11]
Are you sure you're using official FTDI hardware, and not clones?
We had a clone and it stopped working at some point. We ordered some real FTDI ones (which were hardly more expensive) and it worked. Didn't test recently....

**Comment by Jark** [comment]
Yup, a 100% sure. That there's many clones out there which sometimes don't work properly is a well known fact.

This sample was originally specific for the winRT FTDI d2xx driver, that you had to manually install onto Windows IOT on the Raspberry PI and were only able to use through the FTDI library. 

That installation process stopped working after Windows IOT got support for the FTDI chipsets out of the box. Sadly the default apis are not as fast as the FTDI driver direct driver support.

**Comment by mterwoord** [comment]
So that means that the FTDI stuff is still usable, but only using the built-in drivers?

**Comment by Jark** [comment]
You can use this sample with the built-in drivers that support FTDI by uncommenting/commenting the lines at https://github.com/Jark/FTDISample/blob/master/src/FTDISample/MainPage.xaml.cs#L17 

The fast FTDI library from http://www.ftdichip.com/Drivers/D2XX.htm does no longer work so you can't use that approach.

**Comment by mterwoord** [comment]
Cool. Thanks for clarifying: Was scared with the possibility of serial communications not working at all.. 
:)

**Comment by wimdevriendt** [comment]
Hi I have bought a FTDI cable (original one) and when i run the sample i got 
NON GENUINE DEVICE DETECTED
How can i get the sample working ?

**Comment by Jark** [comment]
Hi @wimdevriendt, 

This sample was originally specific for the winRT FTDI d2xx driver, that you had to manually install onto Windows IOT on the Raspberry PI and were only able to use through the FTDI library.

That installation process stopped working after Windows IOT got support for the FTDI chipsets out of the box. 

Microsoft have since then added built-in support for FTDI devices, so if you uncomment / comment the lines at  https://github.com/Jark/FTDISample/blob/master/src/FTDISample/MainPage.xaml.cs#L17 it should work for you.

I'll add it to my todo list to make the SerialDeviceManager the standard way of doing things.

Hope this helps,

Jark

**Comment by wimdevriendt** [comment]
Hi, if i uncomment is nothing is working. I have installed the drivers so maybe i should reimage the windows iot and not copy the drivers ? right ?

**Comment by Jark** [comment]
That might be worth a try, let me know if that fixes it for you.

Unfortunately I don't have any working testing setup, so I can't verify for you if the sample is still working, the last time I tested this sample was around the same time last year.

However, I would expect some very clear exceptions if anything wasn't working :)

**Comment by wimdevriendt** [comment]
Hi I created new image, indeed i see the usb ttl ftdi in devices without installing drivers
but i get device = null on line  var device = await ftManager.OpenByDeviceId(deviceNode.Id);
when i select the device

**Comment by Jark** [comment]
Seems like you need to add some device capabilities to the appxmanifest file to get it to work now:
https://social.msdn.microsoft.com/Forums/en-US/b9633593-377e-4d6f-b3a9-838de0555371/serialdevicefromidasync-always-returns-null-unless-the-serial-adapter-is-plugged-in-after-boot?forum=WindowsIoT

Give that a go.

**Comment by wimdevriendt** [comment]
Also it’s not working
It’s already in the package

**Comment by Jark** [comment]
Just to be clear: the specific DeviceCapability mentioned in that link are _not_ the same as the package manifest file supplied with the sample code in this repository.

If it's still not working it's possible that you've got a revision that's too new / different from the drivers that are supplied with Windows 10 iot or you've got a fake FTDI chip. You would have to either contact Microsoft directly or contact FTDI to find out if your usb serial device is supported or not.
